src/main/java/com/red5pro/group/ICoreProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.red5pro.group;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+/**
+ * Provide logical handlers for Composition objects.
+ * 
+ * @author Andy
+ *
+ */
+public interface ICoreProvider {
+	/**
+	 * Application developers add handlers here.
+	 */
+	public static List<ICoreProvider> registry = new CopyOnWriteArrayList<>();
+	/**
```

Comment:
My concerns here are that theres a static list which is only used for look up `resolve`, but no means to `add` or `remove` elements of said list. Does this list grow over time or is it a static one-shot?

Reply:
ICoreProvider  is a class factory that can resolve classes implementing IGroupCore, The registry is a list of those providers.
The mixer reactor instance which is resolved by an ICoreProvider is not stored in the list since the types dont match.
 'List' contains its own 'add' and 'remove' methods suitable for replacing or removing these class factories. 
 If you are concerned about the disposal of the IGroupCore instance that may be 'resolved', it already has its own lifecycle contract defined.   
 
 As you know, class factories within the lib folder cannot load classes located in a webapp's WEBINF/classes folder at runtime.  So we provide a static method handler interface for end users to be called when the ConferenceGroup is looking up the end user's custom chat or shared object codes.

Reply:
All very good points; A++